src/Dapr.Jobs/DaprJobsServiceException.cs

```diff
@@ -0,0 +1,38 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2024 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+namespace Dapr.Jobs;
+
+/// <summary>
+/// The exception type thrown when an exception is encountered using the Dapr Jobs service.
+/// </summary>
+[Serializable]
+public class DaprJobsServiceException : Exception
+{
+    /// <summary>
+    /// Initializes a new <see cref="DaprJobsServiceException"/> for a non-successful HTTP request.
+    /// </summary>
+    /// <param name="response"></param>
+    public DaprJobsServiceException(HttpResponseMessage? response) : base(FormatExceptionForFailedRequest())
+    {
+        Response = response;
+    }
+    
+    /// <summary>
+    /// Gets the <see cref="HttpResponseMessage"/> of the request that failed. Will be <c>null</c> if the
+    /// failure was not related to an HTTP request or preventing the response from being received.
+    /// </summary>
+    public HttpResponseMessage? Response { get; }
```

Comment:
I don't know how I feel about exposing the `HttpResponseMessage`. While it *is* an error and therefore something of a special case, it would seem to leak a bit of the abstraction of the API we're building.  Would it be better for the exception to require a message that more specifically describes what about the HTTP response it didn't like?

Reply:
Same comment for the `DaprMalformedJobException`, too.

Reply:
Ultimately dropped both exceptions because neither one is referenced anywhere. If we need them later on, we can always add them back.